app/javascript/components/artists/ArtistProfile.jsx

```diff
@@ -76,7 +96,19 @@ class ArtistProfile extends React.Component {
         "X_CSRF-Token": document.getElementsByName("csrf-token")[0].content
       }
     }).then((data) => {
-      window.location = `/artists/` + this.props.artist.id;
+      let works = this.state.works;
+      delete works[work_id];
+      this.setState({ works: works });
```

Comment:
Since you're refreshing the page with `window.location = ...` after updating the artist, you don't need to update state here! State will be updated with another API call in `componentDidMount()`.

Can also remove line 99 and 100 :)